File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/objects/MeetingTest.java
package com.github.dedis.popstellar.model.objects;

import androidx.test.ext.junit.runners.AndroidJUnit4;

import com.github.dedis.popstellar.model.objects.event.EventState;
import com.github.dedis.popstellar.utility.security.Hash;

import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.ArrayList;
import java.util.List;

import static org.junit.Assert.assertEquals;

@RunWith(AndroidJUnit4.class)
public class MeetingTest {

  private static final String LAO_ID = "LAO_ID";
  private static final String ID = "ID";
  private static final String NAME = "MEETING_NAME";
  private static final String LOCATION = "Test Location";
  private static final long CREATION = System.currentTimeMillis() / 1000;
  private static final long START = CREATION + 1;
  private static final long END = START + 1;
  private static final long LAST_MODIFIED = CREATION;
  private static final String MODIFICATION_ID = "MOD_ID";
  private static final List<String> MODIFICATION_SIGNATURES = new ArrayList<>();

  @Test
  public void getCorrectId() {
    assertEquals(
        Hash.hash("M", LAO_ID, Long.toString(CREATION), NAME),
        Meeting.generateCreateMeetingId(LAO_ID, CREATION, NAME));
  }

  @Test
  public void testCreateState() {
    Meeting meeting =
        new Meeting(
            ID,
            NAME,
            CREATION,
            START + 10,
            END + 10,
            LOCATION,
            LAST_MODIFIED,
            MODIFICATION_ID,
            MODIFICATION_SIGNATURES);
    assertEquals(EventState.CREATED, meeting.getState());
  }

  @Test
  public void testOpenState() {
    Meeting meeting =
        new Meeting(
            ID,
            NAME,
            CREATION,
            CREATION,
            END,
            LOCATION,
            LAST_MODIFIED,
            MODIFICATION_ID,
            MODIFICATION_SIGNATURES);
    assertEquals(EventState.OPENED, meeting.getState());
  }

  public void testClosedState() {
    Meeting meeting =
        new Meeting(
            ID,
            NAME,
            CREATION,
            CREATION,
            CREATION,
            LOCATION,
            LAST_MODIFIED,
            MODIFICATION_ID,
            MODIFICATION_SIGNATURES);
    assertEquals(EventState.CLOSED, meeting.getState());
  }
}

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/objects/MeetingTest.kt
package com.github.dedis.popstellar.model.objects

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.github.dedis.popstellar.model.objects.Meeting.Companion.generateCreateMeetingId
import com.github.dedis.popstellar.model.objects.event.EventState
import com.github.dedis.popstellar.utility.security.HashSHA256.hash
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class MeetingTest {
  @Test
  fun correctId() {
    Assert.assertEquals(
      hash("M", LAO_ID, CREATION.toString(), NAME),
      generateCreateMeetingId(LAO_ID, CREATION, NAME)
    )
  }

  @Test
  fun testCreateState() {
    val meeting =
      Meeting(
        ID,
        NAME,
        CREATION,
        START + 10,
        END + 10,
        LOCATION,
        LAST_MODIFIED,
        MODIFICATION_ID,
        MODIFICATION_SIGNATURES
      )
    Assert.assertEquals(EventState.CREATED, meeting.state)
  }

  @Test
  fun testOpenState() {
    val meeting =
      Meeting(
        ID,
        NAME,
        CREATION,
        CREATION,
        END,
        LOCATION,
        LAST_MODIFIED,
        MODIFICATION_ID,
        MODIFICATION_SIGNATURES
      )
    Assert.assertEquals(EventState.OPENED, meeting.state)
  }

  fun testClosedState() {
    val meeting =
      Meeting(
        ID,
        NAME,
        CREATION,
        CREATION,
        CREATION,
        LOCATION,
        LAST_MODIFIED,
        MODIFICATION_ID,
        MODIFICATION_SIGNATURES
      )
    Assert.assertEquals(EventState.CLOSED, meeting.state)
  }

  companion object {
    private const val LAO_ID = "LAO_ID"
    private const val ID = "ID"
    private const val NAME = "MEETING_NAME"
    private const val LOCATION = "Test Location"
    private val CREATION = System.currentTimeMillis() / 1000
    private val START = CREATION + 1
    private val END = START + 1
    private val LAST_MODIFIED = CREATION
    private const val MODIFICATION_ID = "MOD_ID"
    private val MODIFICATION_SIGNATURES: List<String> = ArrayList()
  }
}
